### PR TITLE
feat: support contract declaration API introduced at SN 0.9.0

### PR DESCRIPTION
**Pull request by yoga-braavos**
This PR introduces support for the new 'declare' workflow introduced at Starknet's latest 0.9.0 upgrade.
[Starknet docs](https://docs.starknet.io/docs/Blocks/transactions/#declare-transaction)

### CODE REVIEW

File: src/provider/default.ts [hunk 1]
```diff
@@ -308,6 +309,34 @@ export class Provider implements ProviderInterface {
...
 
+  /**
+   * Declare a given compiled contract (json) on starknet
+   *
+   * @param contract - a json object containing the compiled contract
+   * @returns a confirmation of sending a transaction on the starknet contract
+   */
+  public declareContract(
+    payload: DeclareContractPayload,
+    _abi?: Abi
```
Comment (badurinantun) [inline src/provider/default.ts:320]: Can this be removed?

File: __tests__/provider.test.ts [hunk 2]
```diff
@@ -123,6 +123,16 @@ describe('defaultProvider', () => {
   });
 
   describe('addTransaction()', () => {
+    test('declareContract()', async () => {
+      const response = await defaultProvider.declareContract({
+        contract: compiledErc20,
+      });
+
+      expect(response.code).toBe('TRANSACTION_RECEIVED');
```
[the remark this answers is not in the document]
Comment (yoga-braavos) [reply __tests__/provider.test.ts:131]: It's out of scope for this test as we're not checking for correctness of the Starknet infrastructure but just that our API call was correct and gone through all the relevant sanity checks and eventually returned a `class_hash`

File: src/provider/default.ts (hunk 1 above)
Comment (badurinantun) [reply src/provider/default.ts:320]: I'm not sure why those `_abi`s are there TBH :smile:

**Comment by yoga-braavos** [comment]
rebased on latest `develop`

**Comment by yoga-braavos** [comment]
> rebased on latest `develop`

again. Tests that need private key are failing..

**Pull request merged**